p2p/discovery.py

```diff
@@ -948,21 +953,49 @@ class DiscoveryService(BaseService):
     _last_lookup: float = 0
     _lookup_interval: int = 30
 
-    def __init__(self, proto: DiscoveryProtocol, peer_pool: BasePeerPool,
-                 port: int, token: CancelToken = None) -> None:
+    def __init__(self,
+                 proto: DiscoveryProtocol,
+                 port: int,
+                 event_bus: Endpoint,
+                 token: CancelToken = None) -> None:
         super().__init__(token)
         self.proto = proto
-        self.peer_pool = peer_pool
         self.port = port
+        self._event_bus = event_bus
         self._lookup_running = asyncio.Lock()
 
+        self.run_daemon_task(self.handle_get_peer_candidates_requests())
+        self.run_daemon_task(self.handle_get_random_bootnode_requests())
+
+    async def handle_get_peer_candidates_requests(self) -> None:
+        async for event in self._event_bus.stream(PeerCandidatesRequest):
+
+            # Not sure if we should be doing this here
+            self.run_task(self.maybe_lookup_random_node())
```

Comment:
@pipermerriam a controversial part I wanted to highlight. This is placed more or less randomly here. The idea was to somewhat mimic the old behavior. We could also just do that on a given interval. Not sure.

Reply:
I think this is fine.